src/main/java/com/megastore/facade/ProductFacade.java

```diff
@@ -10,8 +10,18 @@
 public interface ProductFacade {
     ProductPDPDto findById(Long id);
 
-    Collection<ProductPLPDto> findAll();
+    Collection<ProductPLPDto> findAll(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
+    Collection<ProductPLPDto> findAllSortedByNameDESC(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
+    Collection<ProductPLPDto> findAllSortedByNameASC(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
+    Collection<ProductPLPDto> findAllSortedByPriceDESC(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
+    Collection<ProductPLPDto> findAllSortedByPriceASC(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
```

Comment:
There shouldn't be 5 different functions for each sort type. Why not do the same as with filtering parameters?